packages/twenty-front/src/modules/object-metadata/hooks/useGetSyncStatusMetadata.ts

```diff
@@ -0,0 +1,25 @@
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { isUndefinedOrNull } from '~/utils/isUndefinedOrNull';
+
+export const useGetSyncStatusMetadata = () => {
```

Comment:
this should live in calendar front module (or in modules/settings/accounts) (it's not something generic like object-metadata)
other than than it looks good!